crates/via-router/src/path/pattern.rs

```diff
@@ -15,30 +15,13 @@ pub struct Param {
     ident: Box<str>,
 }
 
-/// Unwraps the remaining slice of a path segment after the first char or
-/// panics with a custom message.
-///
-macro_rules! rest_or {
-    (
-        // Should evaluate to a `&str`.
-        $segment:expr,
-        // Args passed to panic! if the segment is empty.
-        $($arg:tt)*
-    ) => {{
-        let segment = $segment;
-        match segment.get(1..) {
-            // The range is out of bounds or produced an empty str.
-            None | Some("") => panic!($($arg)*),
-            // The range is valid.
-            Some(rest) => rest,
-        }
-    }};
-}
-
 /// Returns an iterator that yields a `Pattern` for each segment in the uri path.
 ///
 pub fn patterns(path: &'static str) -> impl Iterator<Item = Pattern> {
```

Comment:
This isn't the most efficient implementation but this isn't a hot code path. This function is not called during a request.